changelog/_unreleased/2024-12-17-refactor-form-handling.md

```diff
@@ -0,0 +1,362 @@
+* Optionally every input field can have a description area beneath it to give more context to the user. If present, the description is also referenced via the `aria-describedby` attribute and read out by screen readers.
+* Besides the normal color feedback, invalid fields now also display an error icon on the right side of the input. This is an important visual feedback for users which find it difficult to identify colors.
+* Placeholder labels were removed from most form fields, as they are irritating and don't add value, especially if they just mirror the content of the label.
+
+#### Form validation
+* Besides the immediate field validation feedback, all fields will be validated when submitting the form. 
+* If there are still invalid fields, they will be highlighted with the necessary visual feedback.
+* In addition, the page will focus the first invalid field. The browser will automatically scroll the page to that field if it is not in view.
+
+### Implementation
+
+#### New validation service and form handler plugin
+There is a new central form validation class that is also available as a default instance under `window.formValidation`. This is used by a new form handler plugin that will automatically implement the necessary events and handling on a form element. It can be activated with the `data-form-handler="true"` attribute on a form element.
+
+**Example:**  
+```HTML
+<form action="/newsletter" method="post" data-form-handler="true">
+    <label for="email">Email</label>
+    <input type="email" id="email" name="email" data-validation="required,email">
+    
+
+    <button type="submit">Submit</button>
+</form>
+```
+The form validation works with an associated `data-validation` attribute on the form fields. You can pass a comma separated list of validator keys. Their priority is defined by their order. Only the validation message of the highest applying validator is shown to the user for relevant feedback.
+
+These validators are available by default:  
+
+| Key      | Description |
+| -------- | ------- |
+| `required` | Checks if the field is not empty. |
+| `email` | Checks the value of the field to be a valid email address. |
+| `conformation` | Checks if the value of a confirmation field matches the value of the original field. Make sure to use the right ID naming for the validator to work. As an example, the orginal field has the ID `email` and the confirmation field has the ID `emailConfirmation`. The `confirmation` validator should be added to the confirmation field. Note that unnecessary inputs are seen as not accessible and should be avoided wherever possible. |
+| `minLength` | Checks the value of the field for a minimum length. If available the validator will use the `minlength` attribute of the field to validate against. Otherwise, it will use the default configuration of eight characters. | 
+
+You can take a look at the reference documentation of the service and the plugin for further information.
```

Comment:
I think the helper `form-validation.helper.js` is meant here? Maybe we can tell the file name here again or even provide a one liner example with:

You can add your own validators using: `window.formValidation.addValidator()`